app/scripts/modules/google/src/serverGroup/configure/wizard/loadBalancingPolicy/loadBalancingPolicySelector.component.html

```diff
@@ -1,19 +1,56 @@
 <ng-form name="loadBalancingPolicySubForm">
   <collapsible-section heading="Port Name Mapping" expanded="true" subsection="true">
     <div class="form-group">
-      <div class="col-md-5 sm-label-right">
-        <span>http</span><help-field key="gce.serverGroup.loadBalancingPolicy.listeningPort"></help-field>
+      <div class="form-group row" ng-repeat="namedPort in $ctrl.command.loadBalancingPolicy.namedPorts">
+        <div class="col-md-12">
+          <div class="col-md-2 sm-label-right">
+            <b>Name</b>
+            <help-field key="gce.serverGroup.loadBalancingPolicy.portName"></help-field>
+          </div>
+          <div class="col-md-3">
+            <ui-select ng-if="!namedPort.name || !namedPort.name.length"
+                       ng-model="namedPort.name"
+                       class="form-control input-sm">
+              <ui-select-match>{{$item.name}}</ui-select-match>
+              <ui-select-choices repeat="portName in $ctrl.getPortNames()">
+                <span ng-bind-html="portName"></span>
+              </ui-select-choices>
+            </ui-select>
+
+            <input ng-if="namedPort.name.length"
+                   ng-readonly="namedPort.name.length"
+                   type="string"
+                   name="portName"
+                   class="form-control input-sm"
+                   ng-model="namedPort.name"/>
+          </div>
+          <div class="col-md-2 sm-label-right">
+            <b>Port</b>
+            <help-field key="gce.serverGroup.loadBalancingPolicy.listeningPort"></help-field>
+          </div>
+          <div class="col-md-3">
+            <input type="number"
+                   name="listeningPort"
+                   required
+                   class="form-control input-sm"
+                   ng-model="namedPort.port"
+                   max="{{ $ctrl.maxPort }}"
+                   min="1"/>
+          </div>
+          <div class="col-md-2 sm-label-left">
+            <span class="glyphicon glyphicon-trash" uib-tooltip="Remove Named Port"
+                  ng-click="$ctrl.removeNamedPort(namedPort)"></span>
```

Comment:
Inside an `ng-repeat`, you can reference `$index`. If you pass `$index` to `$ctrl.removeNamedPort`, you can rewrite `$ctrl.removeNamedPort`:

```
public removeNamedPort(index: number): void {
    this.command.loadBalancingPolicy.namedPorts.splice(index, 1);
}
```

This also helps you avoid some nasty problems that would show up if the same named port was listed twice (which  probably shouldn't happen, and ideally would be validated).